lib/IPC/LeaderBoard.pm

```diff
@@ -366,7 +376,9 @@ sub update {
     if (%private_values) {
         my $idx_delta = $self->_generation_idx + 1;
         while (my ($private_idx, $value) = each %private_values) {
```

Comment:
Please avoid `each` here or reset the iterator explicitly before the loop. The reason is because the loop can be exited via non-local goto before the iterator ran out.